awswrangler/s3/_read_parquet.py

```diff
@@ -236,457 +216,269 @@ def _read_parquet_metadata(
+
+    if chunked:
+        return _read_parquet_chunked(
             boto3_session=boto3_session,
-            s3_additional_kwargs=s3_additional_kwargs,
+            paths=paths,
+            path_root=path_root,
+            columns=columns,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            chunked=chunked,
             use_threads=use_threads,
-            version_id=version_id,
-            pyarrow_additional_kwargs=pyarrow_args,
-        ),
-        categories=categories,
-        safe=safe,
-        map_types=map_types,
-        use_threads=use_threads,
-        dataset=dataset,
-        path=path,
-        path_root=path_root,
-        timestamp_as_object=pyarrow_args["timestamp_as_object"],
+            s3_additional_kwargs=s3_additional_kwargs,
+            arrow_kwargs=arrow_kwargs,
+            version_ids=version_ids,
+        )
+
+    executor = _get_executor(use_threads=use_threads)
+    tables = executor.map(
+        _read_parquet_file,
+        boto3_session,
+        paths,
+        itertools.repeat(path_root),
+        itertools.repeat(columns),
+        itertools.repeat(coerce_int96_timestamp_unit),
+        itertools.repeat(s3_additional_kwargs),
+        itertools.repeat(use_threads),
+        [version_ids.get(p) if isinstance(version_ids, dict) else None for p in paths],
     )
-    if validate_schema and columns:
-        for column in columns:
-            if column not in df.columns:
-                raise exceptions.InvalidArgument(f"column: {column} does not exist")
-    return df
+    return _utils.table_refs_to_df(tables=tables, kwargs=arrow_kwargs)
 
 
 def read_parquet(
     path: Union[str, List[str]],
     path_root: Optional[str] = None,
+    dataset: bool = False,
     path_suffix: Union[str, List[str], None] = None,
     path_ignore_suffix: Union[str, List[str], None] = None,
-    version_id: Optional[Union[str, Dict[str, str]]] = None,
     ignore_empty: bool = True,
-    ignore_index: Optional[bool] = None,
     partition_filter: Optional[Callable[[Dict[str, str]], bool]] = None,
     columns: Optional[List[str]] = None,
     validate_schema: bool = False,
-    chunked: Union[bool, int] = False,
-    dataset: bool = False,
-    categories: Optional[List[str]] = None,
-    safe: bool = True,
-    map_types: bool = True,
-    use_threads: Union[bool, int] = True,
+    coerce_int96_timestamp_unit: Optional[str] = None,
     last_modified_begin: Optional[datetime.datetime] = None,
     last_modified_end: Optional[datetime.datetime] = None,
+    version_id: Optional[Union[str, Dict[str, str]]] = None,
+    chunked: Union[bool, int] = False,
+    use_threads: Union[bool, int] = True,
+    parallelism: int = 200,
     boto3_session: Optional[boto3.Session] = None,
     s3_additional_kwargs: Optional[Dict[str, Any]] = None,
-    pyarrow_additional_kwargs: Optional[Dict[str, Any]] = None,
+    arrow_additional_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
Nitpick - I'd rather not rename that :) Not sure what is the benefit

Reply:
I am not fussed about the actual name tbh, it's just that it's a bit inconsistent at the moment. For [s3 select](https://github.com/awslabs/aws-data-wrangler/blob/release-3.0.0/awswrangler/s3/_select.py#L144), it's `arrow_additional_kwargs`, for athena it's `pyarrow_additional_kwargs`. As long as we stick to one I am happy...